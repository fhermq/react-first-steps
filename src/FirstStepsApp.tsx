import { ItemCounter } from "./shopping-cart/ItemCounter";

interface ItemInCart {
    productName: string;
    quantity: number;
}
const itemsInCart: ItemInCart[] = [
    { productName: 'Nintendo Switch 2', quantity: 1 },
    { productName: 'Pro Controller', quantity: 5 },
    { productName: 'Super Smash Bros', quantity: 1 },
    { productName: 'Memoria SD 256 GB', quantity: 3 },
]

export function FirstStepsApp() {
    return (

        // <>
        //     <h1>Hola Mundo!</h1>
        //     <p>Esto es p√°rrafo</p>
        //     <button>Click me</button>
        //     <div>
        //         <h2>Hola dentro de un div</h2>
        //     </div>
        // </>
        <>
            <h1 data-testid="test-h1">Carrito de compras</h1>
            {/* <ItemCounter name="Nintendo Switch 2" quantity={1}/>
            <ItemCounter name="Pro Controller" quantity={2} />
            <ItemCounter name="Super Smash Bros" quantity={1} />
            <ItemCounter name="Memoria SD 256 GB" quantity={2} /> */}
        {
            itemsInCart.map( ({productName, quantity})=> (
                 <ItemCounter key={productName} name={productName} quantity={quantity} />
            ))
        }

        </>
    );
}