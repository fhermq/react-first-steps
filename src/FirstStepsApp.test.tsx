import { describe, test, expect } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";
import { render, screen } from '@testing-library/react'


describe('FirstStepsApp', () =>{

    test('should match snapshot', ()=>{
        render(<FirstStepsApp/>);
        expect( screen.getByTestId('test-h1')).toMatchSnapshot();
    } );

    test('should match snapshot with container', ()=>{
       const {container} = render(<FirstStepsApp/>);
        expect(container).toMatchSnapshot();
        screen.debug(); 
    } );

})