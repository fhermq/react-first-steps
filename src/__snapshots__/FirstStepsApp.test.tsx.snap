// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
<h1
  data-testid="test-h1"
>
  Carrito de compras
</h1>
`;

exports[`FirstStepsApp > should match snapshot with container 1`] = `
<div>
  <h1
    data-testid="test-h1"
  >
    Carrito de compras
  </h1>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Nintendo Switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      1
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Pro Controller
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Super Smash Bros
    </span>
    <button>
      +1
    </button>
    <span>
      1
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Memoria SD 256 GB
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
